### ISSUE TITLE
Scenario selector - The end user should be able to filter scenarios

### ISSUE THREAD

**Issue by AlexandreSajus** [Avaiga/taipy#1075]
Here's an example: I am doing a what-if analysis on a neural network by changing the parameter alpha. When I create a new scenario, "alpha 2.00", it gets put at the top of the list instead of being ordered alphabetically. The end user should be able to filter scenarios in the scenario_selector.

![core_issues_1_1](https://github.com/Avaiga/taipy/assets/62465003/3c1de549-ac37-436f-80d8-6c992615ccbe)

**Comment by jrobinAV** [comment]
Related tickets:
https://github.com/Avaiga/taipy/issues/169
https://github.com/Avaiga/taipy/issues/1040
https://github.com/Avaiga/taipy/issues/392

**Comment by bobbyshermi** [comment]
I believe my request is close to this issue. If it is not, please tell me so I can create another issue. My request would be to have better tools to navigate between scenarios. A sort and a filter could definitely help, but a search bar as well, or the possibility to oversee the tags directly from the selector.

**Use case:** 
I am using taipy for finding suspicious patterns in multiple datasets. Every day I have a team of users creating a few scenarios per dataset. I get between 20 to 80 scenarios a day. And I need to review them all within a few days. This is crucial to me to be able to rapidly find a scenario within a large number of scenarios. In my case, the scenario selector cannot be used anymore after a couple of days.

**Workaround**
I had to do many things: 
* First I had to enforce a naming convention. My team as a high turn over rate, and some users are participating for free to the exerise only once. So it is cumbersome to ask them to apply such scenario naming rules.
* I had to design a full page with tens of widgets ( inputs, text, selectors, etc. ) to have a proper way of searching scenarios. I am using some dates, labels, kpis, comments, and dataset names to search for my scenarios.
* I am archiving some scenarios downloading the data not to lose the results. Then I created another page to access the archived datasets and results. This is not so important since I just need to use this page a couple of times a year.

Please tell me if I need to create a new issue for that.

**Comment by FredLL-Avaiga** [comment]
[Specs](https://docs.google.com/document/d/19R0SiykrlGALWU9ONDdtmni4PNBmJ-95s9i755FhWgg/edit#heading=h.pd8hgjgu53fj)

**Comment by bobbyshermi** [comment]
Do you have any target date for a release?

**Comment by jrobinAV** [comment]
> Do you have any target date for a release?

@bobbyshermi,
Thank you for your interest and contribution to Taipy.

As you know, we are currently working on multiple issues to improve the scenario selector user experience. All the issues will be addressed in the next minor release, but we haven't communicated a specific date. We will do our best to make it for the next few months. That's all we can say.

Even if it is a non-stable nor definitive version, you can test what is already implemented in the develop branch of taipy. Feel free to give us your feedback and thoughts.

**Comment by bobbyshermi** [comment]
@jrobinAV 

All right, I 'll have a look to the develop version.

Thanks

**Comment by FlorianJacta** [comment]
Just so you know, it is currently being tested, so there might still be issues
